### DISCUSSION TITLE
IOS not rendering GLB neither usdz

### DISCUSSION THREAD

**Discussion by artsmorgan** [google/model-viewer#4152] [Q&A]
Hi guys,

I'm trying to use the AR feature and everything seems to be working in desktop, but when I switch to mobile nothing works.

I have this simple HTML boiler plate with the model-view js file and my object

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>HTML 5 Boilerplate</title>
    <link rel="stylesheet" href="style.css">
  </head>
  <body>
    <model-viewer src="assets/idle.glb" ar ar-scale="fixed" camera-controls touch-action="pan-y" alt="A 3D model of an astronaut" ios-src="assets/idle.usdz" xr-environment></model-viewer>
    <script type="module" src="js/model-viewer.min.js"></script>
  </body>
</html>
```

And when debugging in chrome got this error:

Warning: TypeError: null is not and object (evaluating 't.getShaderPrecisionFormat (35633, 36338).pression')
LOG: THREE.WebGLRendered: Context Lost.

Any Idea how can I make it work?

**Comment by danilokrulj** [comment]
I think you are missing this: ar-modes="webxr scene-viewer quick-look"

**Reply by artsmorgan** [reply]
Thanks @danilokrulj , same result, the mobile browser crashes.

**Reply by elalish** [reply]
@danilokrulj those are our default AR modes now, so that shouldn't matter. @artsmorgan What version of MV and what device and browser are you using?

**Reply by artsmorgan** [reply]
@elalish sorry late.

Tested in:

MV Version: 3.0.1 (the same as the website documentation CDN)

Iphone 13 Pro Max

- Chrome 
- Safari

**Reply by elalish** [reply]
Can you give us a repro link?

**Reply by artyomxx** [reply]
Latest iOS, iPhone 13, cosmonaut on modelviewer.dev can't be seen.

**Reply by elalish** [reply]
Hmm, that sounds not good. Have you filed a bug on WebKit yet? Are you getting the same JS console error as above?
